resource/storage.go

```diff
@@ -8,23 +8,27 @@ import (
 
 // Storer defines the interface of an handler able to store and retrieve resources
 type Storer interface {
-	// Find searches for items in the backend store matching the lookup
-	// argument. The pagination argument must be respected. If no items are
-	// found, an empty list should be returned with no error.
+	// Find searches for items in the backend store matching the q argument. The
+	// Window of the query must be respected. If no items are found, an empty
+	// list should be returned with no error.
 	//
 	// If the total number of item can't be computed for free, ItemList.Total
 	// must be set to -1. Your Storer may implement the Counter interface to let
 	// the user explicitly request the total.
 	//
-	// The whole lookup query must be treated. If a query operation is not
-	// implemented by the storage handler, a resource.ErrNotImplemented must be
-	// returned.
+	// The whole query must be treated. If a query predicate operation or sort
+	// is not implemented by the storage handler, a resource.ErrNotImplemented
+	// must be returned.
+	//
+	// A storer must ignore the Projection part of the query and always return
```

Comment:
I could make this "must" a "should" and explain why. Some storer (like SQL) might want to ignore that recommendation (i.e.: thru an option) to avoid doing a `SELECT *` with a clear statement that it might break interop with some intermediate storers.